### ISSUE TITLE
Numeric Fields - Large Scale Numeric Numbers Not Shown Properly

### ISSUE THREAD

**Issue by mashaalmemon** [jakob/Postico#680]
## What did you do?

Created a numeric field with precision/scale as follows:
```
numeric(51,50)
```
Inserted the following number and saved
```
3.14159265358979323846264338327950288419716939937510
```
Closed the database connection. Then started another one back to the same database and looked at the table/value inserted.

## What did you expect to happen?

I expected to see the following value in the database as inserted:
```
3.14159265358979323846264338327950288419716939937510
```

## What actually happened?

I saw what seems to be a truncated version with the right number of decimal places but wrong number:
```
3.14159265358979323846264338327950288419000000000000
```

Note I tried this with another DB client called "TablePlus" and it worked as expected. 

## What software versions are you using?

Postico version:  
1.5.9 (5273)

macOS version:  
10.14.6

PostgreSQL version:  
9.5

**Comment by jakob** [comment]
Oh, this is bad.

Postico internally converts any number you type into a numeric field into an `NSDecimalNumber`, which is limited to 38 decimal digits of precision. The same happens when displaying data from `NUMERIC` columns.

The same thing happens when displaying data from the server.

I think the best thing to do would be to just have Postico treat all values as strings to avoid the limitations of the Foundation classes (like NSNumber, NSDecimalNumber, etc.).

We do exploit the fact that we use native classes in a few places (eg. when copy/pasting numbers, to convert decimal separators according to region settings for compatibility with other apps).

**Comment by mashaalmemon** [comment]
Ah, yes using any out of the box number library will definately not be able to represent all numbers that can be represented by a PostgreSQL numeric field.

I noticed for the numeric type many PostgreSQL libraries treat numeric fields as strings as you suggested. I concur. 

Let me know when fixed and I can grab that latest version and try again.

**Comment by tsutsu** [comment]
We just hit this same bug a little while ago, in Postico v1.5.17. This is a pretty long-standing bug!

**Comment by jakob** [comment]
Thanks for the reminder, we've now fixed the issue with truncated numbers. It was a bit more elaborate than just using strings, since Postico also has support for sorting SQL query results locally.

In any case, the fix will be released in Postico 1.5.18.

If you have a few moments, could you download this pre-release build and let us know if it resolves the issue for you?

https://eggerapps-downloads.s3.eu-west-1.amazonaws.com/postico-7881.zip

Best regards,
Jakob

**Comment by martinjankoehler** [comment]
Thanks for your report, @mashaalmemon @tsutsu !

This issue (_Numeric Fields - Large Scale Numeric Numbers Not Shown Properly_)
is now fixed in the latest 2.0 beta and today's release 1.5.18 [(download)](https://eggerapps-downloads.s3.eu-west-1.amazonaws.com/postico-1.5.18.zip).

Closing this one now, please feel free to open a new one should any issues occur.
